src/test/java/reposense/wizard/WizardRunnerTest.java

```diff
@@ -0,0 +1,146 @@
+                .build();
+        InputStream targetStream = new ByteArrayInputStream(input.getBytes());
+        Scanner sc = new Scanner(targetStream);
+        wizardRunner.buildInput(sc);
+
+        String expectedInput = new InputBuilder()
+                .addRepos(REPO_LINK)
+                .addUntilDate(UNTIL_DATE)
+                .build();
+        assertEquals(expectedInput, wizardRunner.getBuiltInput());
+    }
+
+    @Test
+    public void buildInput_sinceDateAndUntilDateAndRepo_success() {
+        WizardRunner wizardRunner = new WizardRunner(new BasicWizard());
+        String input = new WizardInputBuilder()
+                .add(REPO_LINK)
+                .addYesFlag()
+                .add(SINCE_DATE)
+                .addYesFlag()
+                .add(UNTIL_DATE)
+                .addNoFlag()
+                .build();
+        InputStream targetStream = new ByteArrayInputStream(input.getBytes());
+        Scanner sc = new Scanner(targetStream);
+        wizardRunner.buildInput(sc);
+
+        String expectedInput = new InputBuilder()
+                .addRepos(REPO_LINK)
+                .addSinceDate(SINCE_DATE)
+                .addUntilDate(UNTIL_DATE)
+                .build();
+        assertEquals(expectedInput, wizardRunner.getBuiltInput());
+    }
+
+    @Test
+    public void buildInput_viewAndRepo_success() {
+        WizardRunner wizardRunner = new WizardRunner(new BasicWizard());
+        // Yes only for view flag
+        String input = new WizardInputBuilder()
+                .add(REPO_LINK)
+                .addNoFlag()
+                .addNoFlag()
+                .addYesFlag()
+                .build();
+        InputStream targetStream = new ByteArrayInputStream(input.getBytes());
+        Scanner sc = new Scanner(targetStream);
+        wizardRunner.buildInput(sc);
+
+        String expectedInput = new InputBuilder()
+                .addRepos(REPO_LINK)
+                .addView()
+                .build();
+        assertEquals(expectedInput, wizardRunner.getBuiltInput());
+    }
+
+    @Test
+    public void buildInput_allFlags_success() {
+        WizardRunner wizardRunner = new WizardRunner(new BasicWizard());
+        // Yes for all flags
+        String input = new WizardInputBuilder()
+                .add(REPO_LINK)
+                .addYesFlag()
+                .add(SINCE_DATE)
+                .addYesFlag()
+                .add(UNTIL_DATE)
+                .addYesFlag()
+                .build();
+        InputStream targetStream = new ByteArrayInputStream(input.getBytes());
+        Scanner sc = new Scanner(targetStream);
+        wizardRunner.buildInput(sc);
+
+        String expectedInput = new InputBuilder()
+                .addRepos(REPO_LINK)
+                .addSinceDate(SINCE_DATE)
+                .addUntilDate(UNTIL_DATE)
+                .addView()
+                .build();
+        assertEquals(expectedInput, wizardRunner.getBuiltInput());
+    }
+}
```

Comment:
I was wondering if it might be better to test individual prompts instead and have a separate test suite for input builder which tests combining multiple repo flags. 
This way, you don't have to rewrite the inputs every time the ordering of prompts is changed.

Additionally, I think it's missing test cases for completely invalid inputs for the optional prompt queries since those are supposed to repeat on invalid inputs if I'm understanding it correctly.

Reply:
We can implement a `TestWizard` (similar to `BasicWizard`) that takes in individual prompts. That way, we can test any individual prompts/combinations of prompts. In this way, we reduce the number of test cases for `BasicWizard` as such, so that the re-writing of input doesn't have to happen for any of the possible new wizards we write, while still being able to test for correctness overall.

Just to check, the test case you mentioned is the one where we repeated input garbage to the OptionalPrompt, and finally a "yes"?

Reply:
Sorry I'm not sure if I'm misunderstanding this 
> Just to check, the test case you mentioned is the one where we repeated input garbage to the OptionalPrompt, and finally a "yes"?

But is that suggesting that such a test case exists already or that that is the test case I want. Because I don't think I see such a test case and yes that's kind of what I want.

In general, I'd like to see more individualized test cases and only a few combined ones that are preferably done in a way that is independent of the order of the actual prompts since those may be reordered arbitrarily and we wouldnt want to have to fix test cases every time we do that. 

Also I guess I have to ask at this point what the difference between TestWizard and Wizard would be

Reply:
> But is that suggesting that such a test case exists already or that that is the test case I want. Because I don't think I see such a test case and yes that's kind of what I want.

Sorry for the confusion, this is suggesting the test case that you want, and does not already exist.

> In general, I'd like to see more individualized test cases and only a few combined ones that are preferably done in a way that is independent of the order of the actual prompts since those may be reordered arbitrarily and we wouldnt want to have to fix test cases every time we do that.

This would be a beneficial change. Though it is as you mentioned, there is still going to be some (but minimal number) of test cases with the ordered prompts.

> Also I guess I have to ask at this point what the difference between TestWizard and Wizard would be

TestWizard would just be a concrete implementation of the abstract Wizard class. For example, we have a BasicWizard currently. TestWizard will be a new, test-specific implementation of the abstract Wizard class where we can input specific types of Prompts specifically to test the prompts within the TestWizard.